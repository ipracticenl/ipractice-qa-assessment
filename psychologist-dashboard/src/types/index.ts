// Base types
export interface User {
  id: number;
  name: string;
  type: "psychologist" | "client";
}

// Time range base type
export interface TimeRange {
  id: string;
  from: Date;
  to: Date;
}

// Client DTOs
export interface CreateClientDto {
  name: string;
  initialPsychologistIds: number[];
}

export interface ClientSummaryDto {
  id: number;
  name: string;
}

export interface ClientDetailsDto {
  id: number;
  name: string;
  assignedPsychologists: number[];
  bookedAppointments: Appointment[];
}

// Psychologist DTOs
export interface CreatePsychologistDto {
  name: string;
  initialClients: number[];
}

export interface PsychologistSummaryDto {
  id: number;
  name: string;
}

export interface PsychologistDetailsDto {
  id: number;
  name: string;
  assignedClients: number[];
  availableTimeSlots: AvailableTimeSlot[];
  bookedAppointments: BookedAppointment[];
}

// Booking DTOs
export interface CreateBookingDto {
  psychologistId: number;
  availableTimeSlotId: string;
  comment?: string;
}

export interface ClientDto {
  clientId: number;
}

export interface TimeSlotDto {
  from: Date;
  to: Date;
}

// Model types
export interface AvailableTimeSlot extends TimeRange {
  // Inherits id, from, to from TimeRange
}

export interface Appointment extends TimeRange {
  psychologistId: number;
  comment?: string;
}

export interface BookedAppointment extends TimeRange {
  clientId: number;
}

// Legacy types for backward compatibility
export interface Psychologist extends User {
  type: "psychologist";
  clients: string[];
}

export interface Client extends User {
  type: "client";
  psychologist?: string;
}

export interface Availability {
  id: string;
  psychologistId: number;
  startTime: Date;
  endTime: Date;
  isBooked: boolean;
  clientId?: string;
  clientName?: string;
}
