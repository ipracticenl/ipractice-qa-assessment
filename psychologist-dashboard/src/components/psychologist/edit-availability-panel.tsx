import { Edit, Trash2 } from "lucide-react";
import { format } from "date-fns";
import type { FC } from "react";
import { Card, CardHeader, CardTitle, CardContent } from "../ui/card";
import {
  TableHeader,
  TableRow,
  TableHead,
  TableBody,
  TableCell,
  Table,
} from "../ui/table";
import { Button } from "../ui/button";
import { AvailableTimeSlot } from "@/types";

interface EditAvailabilityPanelProps {
  userAvailabilities: AvailableTimeSlot[];
  handleEditAvailability: (availability: AvailableTimeSlot) => void;
  onDeleteAvailability: (availabilityId: string) => void;
}

export const EditAvailabilityPanel: FC<EditAvailabilityPanelProps> = ({
  userAvailabilities,
  handleEditAvailability,
  onDeleteAvailability,
}) => {
  return (
    <Card data-testid="edit-availability-panel">
      <CardHeader>
        <CardTitle className="flex items-center gap-2">
          <Edit className="w-5 h-5" />
          Edit Availability
        </CardTitle>
      </CardHeader>
      <CardContent>
        <div className="overflow-x-auto">
          <Table>
            <TableHeader>
              <TableRow>
                <TableHead>Time Slot</TableHead>
                <TableHead>Status</TableHead>
                <TableHead>Actions</TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              {userAvailabilities.length === 0 ? (
                <TableRow>
                  <TableCell
                    colSpan={3}
                    className="text-center py-8 text-gray-500"
                  >
                    No availability slots created yet
                  </TableCell>
                </TableRow>
              ) : (
                userAvailabilities.map((availability) => (
                  <TableRow
                    key={availability.id}
                    data-testid={`availability-row-${availability.id}`}
                  >
                    <TableCell className="font-medium">
                      <div className="flex flex-col sm:flex-row sm:items-center gap-1">
                        <span>{format(availability.from, "EEEE")}</span>
                        <span className="text-sm text-gray-600">
                          {format(availability.from, "HH:mm")} -{" "}
                          {format(availability.to, "HH:mm")}
                        </span>
                      </div>
                    </TableCell>
                    <TableCell>
                      <span
                        className={`inline-flex px-2 py-1 rounded-full text-xs font-medium bg-green-100 text-green-800`}
                        aria-label="This slot is available"
                      >
                        Available
                      </span>
                    </TableCell>
                    <TableCell>
                      <div className="flex flex-col sm:flex-row gap-2">
                        <Button
                          size="sm"
                          variant="outline"
                          onClick={() => handleEditAvailability(availability)}
                          disabled={false}
                          className="min-h-[36px] touch-manipulation"
                          data-testid={`edit-availability-${availability.id}`}
                          aria-label={`Edit availability slot for ${format(
                            availability.from,
                            "EEEE HH:mm"
                          )}`}
                        >
                          <Edit className="w-3 h-3 mr-1" />
                          Edit
                        </Button>
                        <Button
                          size="sm"
                          variant="destructive"
                          onClick={() => onDeleteAvailability(availability.id)}
                          disabled={false}
                          className="min-h-[36px] touch-manipulation"
                          data-testid={`delete-availability-${availability.id}`}
                          aria-label={`Delete availability slot for ${format(
                            availability.from,
                            "EEEE HH:mm"
                          )}`}
                        >
                          <Trash2 className="w-3 h-3 mr-1" />
                          Delete
                        </Button>
                      </div>
                    </TableCell>
                  </TableRow>
                ))
              )}
            </TableBody>
          </Table>
        </div>
      </CardContent>
    </Card>
  );
};
