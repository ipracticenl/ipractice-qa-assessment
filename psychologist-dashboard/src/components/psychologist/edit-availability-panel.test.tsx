import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { EditAvailabilityPanel } from "./edit-availability-panel";
import { AvailableTimeSlot } from "@/types";

// Mock data
const mockAvailabilities: AvailableTimeSlot[] = [
  {
    id: "slot-1",
    from: new Date("2024-01-15T10:00:00"),
    to: new Date("2024-01-15T11:00:00"),
  },
  {
    id: "slot-2",
    from: new Date("2024-01-16T14:00:00"),
    to: new Date("2024-01-16T15:00:00"),
  },
];

describe("EditAvailabilityPanel", () => {
  const mockHandleEditAvailability = vi.fn();
  const mockOnDeleteAvailability = vi.fn();

  describe("Snapshot Tests", () => {
    it("should match snapshot when no availabilities exist", () => {
      // Given: No availabilities are provided
      const emptyAvailabilities: AvailableTimeSlot[] = [];

      // When: The component is rendered with no availabilities
      const { container } = render(
        <EditAvailabilityPanel
          userAvailabilities={emptyAvailabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });

    it("should match snapshot when availabilities exist", () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered with availabilities
      const { container } = render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe("Empty State", () => {
    it("should display empty state message when no availabilities exist", () => {
      // Given: No availabilities are provided
      const emptyAvailabilities: AvailableTimeSlot[] = [];

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={emptyAvailabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The empty state message should be displayed
      expect(
        screen.getByText("No availability slots created yet")
      ).toBeInTheDocument();
    });
  });

  describe("Availabilities Display", () => {
    it("should display availabilities when they exist", () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: Availability rows should be displayed
      expect(screen.getByTestId("availability-row-slot-1")).toBeInTheDocument();
      expect(screen.getByTestId("availability-row-slot-2")).toBeInTheDocument();
    });

    it("should display availability date and time correctly", () => {
      // Given: An availability is provided
      const availabilities = [mockAvailabilities[0]];

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The date and time should be formatted correctly
      expect(screen.getByText("Monday")).toBeInTheDocument();
      expect(screen.getByText("10:00 - 11:00")).toBeInTheDocument();
    });

    it('should display status as "Available" for all slots', () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: All slots should show "Available" status
      const availableStatuses = screen.getAllByText("Available");
      expect(availableStatuses).toHaveLength(2);
    });

    it("should have correct aria-label for status badges", () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: Status badges should have correct aria-label
      const statusBadges = screen.getAllByLabelText("This slot is available");
      expect(statusBadges).toHaveLength(2);
    });
  });

  describe("Edit Functionality", () => {
    it("should call handleEditAvailability when edit button is clicked", () => {
      // Given: An availability is provided and the edit function is mocked
      const availabilities = [mockAvailabilities[0]];
      const mockEditFunction = vi.fn();

      // When: The component is rendered and edit button is clicked
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockEditFunction}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      const editButton = screen.getByTestId("edit-availability-slot-1");
      fireEvent.click(editButton);

      // Then: The edit function should be called with the correct availability
      expect(mockEditFunction).toHaveBeenCalledWith(mockAvailabilities[0]);
    });

    it("should have correct aria-label for edit buttons", () => {
      // Given: An availability is provided
      const availabilities = [mockAvailabilities[0]];

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The edit button should have the correct aria-label
      const editButton = screen.getByTestId("edit-availability-slot-1");
      expect(editButton).toHaveAttribute("aria-label");
      expect(editButton.getAttribute("aria-label")).toContain(
        "Edit availability slot for Monday 10:00"
      );
    });
  });

  describe("Delete Functionality", () => {
    it("should call onDeleteAvailability when delete button is clicked", () => {
      // Given: An availability is provided and the delete function is mocked
      const availabilities = [mockAvailabilities[0]];
      const mockDeleteFunction = vi.fn();

      // When: The component is rendered and delete button is clicked
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockDeleteFunction}
        />
      );

      const deleteButton = screen.getByTestId("delete-availability-slot-1");
      fireEvent.click(deleteButton);

      // Then: The delete function should be called with the correct availability ID
      expect(mockDeleteFunction).toHaveBeenCalledWith("slot-1");
    });

    it("should have correct aria-label for delete buttons", () => {
      // Given: An availability is provided
      const availabilities = [mockAvailabilities[0]];

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The delete button should have the correct aria-label
      const deleteButton = screen.getByTestId("delete-availability-slot-1");
      expect(deleteButton).toHaveAttribute("aria-label");
      expect(deleteButton.getAttribute("aria-label")).toContain(
        "Delete availability slot for Monday 10:00"
      );
    });
  });

  describe("Component Structure", () => {
    it("should render the main panel with correct test ID", () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The main panel should be rendered with correct test ID
      expect(screen.getByTestId("edit-availability-panel")).toBeInTheDocument();
    });

    it("should render table headers correctly", () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The table headers should be displayed
      expect(screen.getByText("Time Slot")).toBeInTheDocument();
      expect(screen.getByText("Status")).toBeInTheDocument();
      expect(screen.getByText("Actions")).toBeInTheDocument();
    });

    it("should display the correct title", () => {
      // Given: Availabilities are provided
      const availabilities = mockAvailabilities;

      // When: The component is rendered
      render(
        <EditAvailabilityPanel
          userAvailabilities={availabilities}
          handleEditAvailability={mockHandleEditAvailability}
          onDeleteAvailability={mockOnDeleteAvailability}
        />
      );

      // Then: The title should be displayed
      expect(screen.getByText("Edit Availability")).toBeInTheDocument();
    });
  });
});
