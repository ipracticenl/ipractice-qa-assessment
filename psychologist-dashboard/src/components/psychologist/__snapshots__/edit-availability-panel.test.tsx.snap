// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditAvailabilityPanel > Snapshot Tests > should match snapshot when availabilities exist 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="edit-availability-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-square-pen w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
          />
        </svg>
        Edit Availability
      </div>
    </div>
    <div
      class="p-6 pt-0"
    >
      <div
        class="overflow-x-auto"
      >
        <div
          class="relative w-full overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <thead
              class="[&_tr]:border-b"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Time Slot
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Status
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                data-testid="availability-row-slot-1"
              >
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
                >
                  <div
                    class="flex flex-col sm:flex-row sm:items-center gap-1"
                  >
                    <span>
                      Monday
                    </span>
                    <span
                      class="text-sm text-gray-600"
                    >
                      10:00
                       -
                       
                      11:00
                    </span>
                  </div>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <span
                    aria-label="This slot is available"
                    class="inline-flex px-2 py-1 rounded-full text-xs font-medium bg-green-100 text-green-800"
                  >
                    Available
                  </span>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <div
                    class="flex flex-col sm:flex-row gap-2"
                  >
                    <button
                      aria-label="Edit availability slot for Monday 10:00"
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 min-h-[36px] touch-manipulation"
                      data-testid="edit-availability-slot-1"
                    >
                      <svg
                        class="lucide lucide-square-pen w-3 h-3 mr-1"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                        />
                        <path
                          d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                        />
                      </svg>
                      Edit
                    </button>
                    <button
                      aria-label="Delete availability slot for Monday 10:00"
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-destructive text-destructive-foreground hover:bg-destructive/90 h-9 rounded-md px-3 min-h-[36px] touch-manipulation"
                      data-testid="delete-availability-slot-1"
                    >
                      <svg
                        class="lucide lucide-trash2 w-3 h-3 mr-1"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                      Delete
                    </button>
                  </div>
                </td>
              </tr>
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                data-testid="availability-row-slot-2"
              >
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
                >
                  <div
                    class="flex flex-col sm:flex-row sm:items-center gap-1"
                  >
                    <span>
                      Tuesday
                    </span>
                    <span
                      class="text-sm text-gray-600"
                    >
                      14:00
                       -
                       
                      15:00
                    </span>
                  </div>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <span
                    aria-label="This slot is available"
                    class="inline-flex px-2 py-1 rounded-full text-xs font-medium bg-green-100 text-green-800"
                  >
                    Available
                  </span>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <div
                    class="flex flex-col sm:flex-row gap-2"
                  >
                    <button
                      aria-label="Edit availability slot for Tuesday 14:00"
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 min-h-[36px] touch-manipulation"
                      data-testid="edit-availability-slot-2"
                    >
                      <svg
                        class="lucide lucide-square-pen w-3 h-3 mr-1"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                        />
                        <path
                          d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                        />
                      </svg>
                      Edit
                    </button>
                    <button
                      aria-label="Delete availability slot for Tuesday 14:00"
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-destructive text-destructive-foreground hover:bg-destructive/90 h-9 rounded-md px-3 min-h-[36px] touch-manipulation"
                      data-testid="delete-availability-slot-2"
                    >
                      <svg
                        class="lucide lucide-trash2 w-3 h-3 mr-1"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                      Delete
                    </button>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditAvailabilityPanel > Snapshot Tests > should match snapshot when no availabilities exist 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="edit-availability-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-square-pen w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
          />
        </svg>
        Edit Availability
      </div>
    </div>
    <div
      class="p-6 pt-0"
    >
      <div
        class="overflow-x-auto"
      >
        <div
          class="relative w-full overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <thead
              class="[&_tr]:border-b"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Time Slot
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Status
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0 text-center py-8 text-gray-500"
                  colspan="3"
                >
                  No availability slots created yet
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
